<!DOCTYPE html
        PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<html>
<head>
    <title>BRESEQ :: Summary Statistics</title>
    <style type="text/css">
        body {
            font-family: sans-serif;
            font-size: 11pt;
        }

        th {
            background-color: rgb(0, 0, 0);
            color: rgb(255, 255, 255);
        }

        table {
            background-color: rgb(1, 0, 0);
            color: rgb(0, 0, 0);
        }

        tr {
            background-color: rgb(255, 255, 255);
        }

        .mutation_in_codon {
            color: red;
            text-decoration: underline;
        }

        .snp_type_synonymous {
            color: green;
        }

        .snp_type_nonsynonymous {
            color: blue;
        }

        .snp_type_nonsense {
            color: red;
        }

        .mutation_header_row {
            background-color: rgb(0, 130, 0);
        }

        .read_alignment_header_row {
            background-color: rgb(255, 0, 0);
        }

        .missing_coverage_header_row {
            background-color: rgb(0, 100, 100);
        }

        .new_junction_header_row {
            background-color: rgb(0, 0, 155);
        }

        .copy_number_header_row {
            background-color: rgb(153, 102, 0);
        }

        .alternate_table_row_0 {
            background-color: rgb(255, 255, 255);
        }

        .alternate_table_row_1 {
            background-color: rgb(235, 235, 235);
        }

        .gray_table_row {
            background-color: rgb(230, 230, 245);
        }

        .polymorphism_table_row {
            background-color: rgb(160, 255, 160);
        }

        .highlight_table_row {
            background-color: rgb(192, 255, 255);
        }

        .reject_table_row {
            background-color: rgb(255, 200, 165);
        }

        .user_defined_table_row {
            background-color: rgb(255, 255, 0);
        }

        .information_table_row {
            background-color: rgb(200, 255, 255);
        }

        .junction_repeat {
            background-color: rgb(255, 165, 0)
        }

        .junction_gene {
        }

        .hidden {
            display: none;
        }

        .unhidden {
            display: block;
        }

    </style>
    <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/>
    <script type="text/javascript">
        function hideTog(divID) {
            var item = document.getElementById(divID);
            if (item) {
                item.className = (item.className == 'hidden') ? 'unhidden' : 'hidden';
            }
        }

        function showTog(butID) {
            var button = document.getElementById(butID);
            if (button) {
                button.value = (button.value == 'Show') ? 'Hide' : 'Show';
            }
        }
    </script>

</head>
<body>
<table width="100%" border="0" cellspacing="0" cellpadding="3">
    <tr>
        <td><a href="http://barricklab.org/breseq"><img src="evidence/breseq_small.png"/></a></td>
        <td width="100%">
            <b><i>breseq</i></b>&nbsp;&nbsp;version 0.32.1&nbsp;&nbsp;
            <br><a href="index.html">mutation predictions</a> |
            <a href="marginal.html">marginal predictions</a> |
            <a href="summary.html">summary statistics</a> |
            <a href="output.gd">genome diff</a> |
            <a href="log.txt">command line log</a>
        </td>
    </tr>
</table>

<p>
<h2>Read File Information</h2>
<table border="0" cellspace="1" cellpadding="5">
    <tr>
        <th></th>
        <th>read file</th>
        <th>reads</th>
        <th>bases</th>
        <th>passed&nbsp;filters</th>
        <th>average</th>
        <th>longest</th>
        <th>mapped</th>
    </tr>
    <tr>
        <td><a href="calibration/AU1064.forward.trimmed.paired.error_rates.pdf">errors</a></td>
        <td>AU1064.forward.trimmed.paired</td>
        <td align="right">1,485,637</td>
        <td align="right">201,157,367</td>
        <td align="right">100.0%</td>
        <td align="right">135.4&nbsp;bases</td>
        <td align="right">151&nbsp;bases</td>
        <td align="right">97.7%</td>
    </tr>
    <tr>
        <td><a href="calibration/AU1064.reverse.trimmed.paired.error_rates.pdf">errors</a></td>
        <td>AU1064.reverse.trimmed.paired</td>
        <td align="right">1,485,203</td>
        <td align="right">198,964,012</td>
        <td align="right">100.0%</td>
        <td align="right">134.0&nbsp;bases</td>
        <td align="right">151&nbsp;bases</td>
        <td align="right">96.8%</td>
    </tr>
    <tr class="highlight_table_row">
        <td></td>
        <td><b>total</b></td>
        <td align="right"><b>2,970,840</b></td>
        <td align="right"><b>400,121,379</b></td>
        <td align="right">100.0%</td>
        <td align="right">134.7&nbsp;bases</td>
        <td><b>151</b>&nbsp;bases</td>
        <td align="right">97.2%</td>
    </tr>
</table>
<h2>Reference Sequence Information</h2>
<p>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th></th>
        <th></th>
        <th>seq id</th>
        <th>length</th>
        <th align="center">fit mean</th>
        <th align="center">fit dispersion</th>
        <th align="center">% mapped reads</th>
        <th align="left">description</th>
    </tr>
    <tr>
        <td><a href="evidence/contig00001.overview.png">coverage</a></td>
        <td><a href="calibration/0.unique_coverage.pdf">distribution</a></td>
        <td>contig00001</td>
        <td align="right">318,930</td>
        <td align="center">53.7</td>
        <td align="center">5.6</td>
        <td align="center">4.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00002.overview.png">coverage</a></td>
        <td><a href="calibration/1.unique_coverage.pdf">distribution</a></td>
        <td>contig00002</td>
        <td align="right">275,319</td>
        <td align="center">48.4</td>
        <td align="center">4.3</td>
        <td align="center">3.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00003.overview.png">coverage</a></td>
        <td><a href="calibration/2.unique_coverage.pdf">distribution</a></td>
        <td>contig00003</td>
        <td align="right">261,219</td>
        <td align="center">50.3</td>
        <td align="center">5.0</td>
        <td align="center">3.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00004.overview.png">coverage</a></td>
        <td><a href="calibration/3.unique_coverage.pdf">distribution</a></td>
        <td>contig00004</td>
        <td align="right">252,231</td>
        <td align="center">53.1</td>
        <td align="center">5.4</td>
        <td align="center">3.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00005.overview.png">coverage</a></td>
        <td><a href="calibration/4.unique_coverage.pdf">distribution</a></td>
        <td>contig00005</td>
        <td align="right">241,072</td>
        <td align="center">52.7</td>
        <td align="center">4.5</td>
        <td align="center">3.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00006.overview.png">coverage</a></td>
        <td><a href="calibration/5.unique_coverage.pdf">distribution</a></td>
        <td>contig00006</td>
        <td align="right">240,180</td>
        <td align="center">50.4</td>
        <td align="center">5.0</td>
        <td align="center">3.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00007.overview.png">coverage</a></td>
        <td><a href="calibration/6.unique_coverage.pdf">distribution</a></td>
        <td>contig00007</td>
        <td align="right">238,018</td>
        <td align="center">54.1</td>
        <td align="center">5.0</td>
        <td align="center">3.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00008.overview.png">coverage</a></td>
        <td><a href="calibration/7.unique_coverage.pdf">distribution</a></td>
        <td>contig00008</td>
        <td align="right">219,575</td>
        <td align="center">55.9</td>
        <td align="center">5.7</td>
        <td align="center">3.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00009.overview.png">coverage</a></td>
        <td><a href="calibration/8.unique_coverage.pdf">distribution</a></td>
        <td>contig00009</td>
        <td align="right">197,432</td>
        <td align="center">55.9</td>
        <td align="center">6.0</td>
        <td align="center">2.8%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00010.overview.png">coverage</a></td>
        <td><a href="calibration/9.unique_coverage.pdf">distribution</a></td>
        <td>contig00010</td>
        <td align="right">185,361</td>
        <td align="center">55.5</td>
        <td align="center">5.3</td>
        <td align="center">2.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00011.overview.png">coverage</a></td>
        <td><a href="calibration/10.unique_coverage.pdf">distribution</a></td>
        <td>contig00011</td>
        <td align="right">185,173</td>
        <td align="center">57.0</td>
        <td align="center">5.3</td>
        <td align="center">2.7%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00012.overview.png">coverage</a></td>
        <td><a href="calibration/11.unique_coverage.pdf">distribution</a></td>
        <td>contig00012</td>
        <td align="right">182,169</td>
        <td align="center">65.1</td>
        <td align="center">6.7</td>
        <td align="center">3.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00013.overview.png">coverage</a></td>
        <td><a href="calibration/12.unique_coverage.pdf">distribution</a></td>
        <td>contig00013</td>
        <td align="right">164,728</td>
        <td align="center">48.8</td>
        <td align="center">3.7</td>
        <td align="center">2.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00014.overview.png">coverage</a></td>
        <td><a href="calibration/13.unique_coverage.pdf">distribution</a></td>
        <td>contig00014</td>
        <td align="right">160,584</td>
        <td align="center">60.4</td>
        <td align="center">5.6</td>
        <td align="center">2.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00015.overview.png">coverage</a></td>
        <td><a href="calibration/14.unique_coverage.pdf">distribution</a></td>
        <td>contig00015</td>
        <td align="right">156,343</td>
        <td align="center">59.7</td>
        <td align="center">5.2</td>
        <td align="center">2.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00016.overview.png">coverage</a></td>
        <td><a href="calibration/15.unique_coverage.pdf">distribution</a></td>
        <td>contig00016</td>
        <td align="right">147,287</td>
        <td align="center">49.4</td>
        <td align="center">4.8</td>
        <td align="center">1.9%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00017.overview.png">coverage</a></td>
        <td><a href="calibration/16.unique_coverage.pdf">distribution</a></td>
        <td>contig00017</td>
        <td align="right">145,625</td>
        <td align="center">54.0</td>
        <td align="center">5.3</td>
        <td align="center">2.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00018.overview.png">coverage</a></td>
        <td><a href="calibration/17.unique_coverage.pdf">distribution</a></td>
        <td>contig00018</td>
        <td align="right">136,195</td>
        <td align="center">50.7</td>
        <td align="center">4.9</td>
        <td align="center">1.8%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00019.overview.png">coverage</a></td>
        <td><a href="calibration/18.unique_coverage.pdf">distribution</a></td>
        <td>contig00019</td>
        <td align="right">134,180</td>
        <td align="center">59.3</td>
        <td align="center">5.3</td>
        <td align="center">2.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00020.overview.png">coverage</a></td>
        <td><a href="calibration/19.unique_coverage.pdf">distribution</a></td>
        <td>contig00020</td>
        <td align="right">131,471</td>
        <td align="center">52.5</td>
        <td align="center">5.9</td>
        <td align="center">1.7%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00021.overview.png">coverage</a></td>
        <td><a href="calibration/20.unique_coverage.pdf">distribution</a></td>
        <td>contig00021</td>
        <td align="right">129,637</td>
        <td align="center">54.7</td>
        <td align="center">5.2</td>
        <td align="center">1.8%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00022.overview.png">coverage</a></td>
        <td><a href="calibration/21.unique_coverage.pdf">distribution</a></td>
        <td>contig00022</td>
        <td align="right">127,836</td>
        <td align="center">53.8</td>
        <td align="center">5.5</td>
        <td align="center">1.7%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00023.overview.png">coverage</a></td>
        <td><a href="calibration/22.unique_coverage.pdf">distribution</a></td>
        <td>contig00023</td>
        <td align="right">127,187</td>
        <td align="center">53.3</td>
        <td align="center">4.3</td>
        <td align="center">1.7%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00024.overview.png">coverage</a></td>
        <td><a href="calibration/23.unique_coverage.pdf">distribution</a></td>
        <td>contig00024</td>
        <td align="right">114,278</td>
        <td align="center">51.4</td>
        <td align="center">4.1</td>
        <td align="center">1.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00025.overview.png">coverage</a></td>
        <td><a href="calibration/24.unique_coverage.pdf">distribution</a></td>
        <td>contig00025</td>
        <td align="right">107,199</td>
        <td align="center">51.9</td>
        <td align="center">4.7</td>
        <td align="center">1.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00026.overview.png">coverage</a></td>
        <td><a href="calibration/25.unique_coverage.pdf">distribution</a></td>
        <td>contig00026</td>
        <td align="right">107,112</td>
        <td align="center">54.5</td>
        <td align="center">5.0</td>
        <td align="center">1.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00027.overview.png">coverage</a></td>
        <td><a href="calibration/26.unique_coverage.pdf">distribution</a></td>
        <td>contig00027</td>
        <td align="right">91,015</td>
        <td align="center">53.1</td>
        <td align="center">3.8</td>
        <td align="center">1.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00028.overview.png">coverage</a></td>
        <td><a href="calibration/27.unique_coverage.pdf">distribution</a></td>
        <td>contig00028</td>
        <td align="right">90,804</td>
        <td align="center">70.6</td>
        <td align="center">5.1</td>
        <td align="center">1.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00029.overview.png">coverage</a></td>
        <td><a href="calibration/28.unique_coverage.pdf">distribution</a></td>
        <td>contig00029</td>
        <td align="right">88,731</td>
        <td align="center">49.6</td>
        <td align="center">6.4</td>
        <td align="center">1.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00030.overview.png">coverage</a></td>
        <td><a href="calibration/29.unique_coverage.pdf">distribution</a></td>
        <td>contig00030</td>
        <td align="right">87,478</td>
        <td align="center">58.5</td>
        <td align="center">7.2</td>
        <td align="center">1.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00031.overview.png">coverage</a></td>
        <td><a href="calibration/30.unique_coverage.pdf">distribution</a></td>
        <td>contig00031</td>
        <td align="right">86,024</td>
        <td align="center">53.7</td>
        <td align="center">4.6</td>
        <td align="center">1.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00032.overview.png">coverage</a></td>
        <td><a href="calibration/31.unique_coverage.pdf">distribution</a></td>
        <td>contig00032</td>
        <td align="right">85,121</td>
        <td align="center">50.7</td>
        <td align="center">4.6</td>
        <td align="center">1.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00033.overview.png">coverage</a></td>
        <td><a href="calibration/32.unique_coverage.pdf">distribution</a></td>
        <td>contig00033</td>
        <td align="right">83,705</td>
        <td align="center">54.2</td>
        <td align="center">6.6</td>
        <td align="center">1.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00034.overview.png">coverage</a></td>
        <td><a href="calibration/33.unique_coverage.pdf">distribution</a></td>
        <td>contig00034</td>
        <td align="right">82,070</td>
        <td align="center">69.8</td>
        <td align="center">7.0</td>
        <td align="center">1.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00035.overview.png">coverage</a></td>
        <td><a href="calibration/34.unique_coverage.pdf">distribution</a></td>
        <td>contig00035</td>
        <td align="right">79,069</td>
        <td align="center">58.7</td>
        <td align="center">4.0</td>
        <td align="center">1.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00036.overview.png">coverage</a></td>
        <td><a href="calibration/35.unique_coverage.pdf">distribution</a></td>
        <td>contig00036</td>
        <td align="right">78,663</td>
        <td align="center">52.9</td>
        <td align="center">5.0</td>
        <td align="center">1.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00037.overview.png">coverage</a></td>
        <td><a href="calibration/36.unique_coverage.pdf">distribution</a></td>
        <td>contig00037</td>
        <td align="right">78,540</td>
        <td align="center">53.7</td>
        <td align="center">4.0</td>
        <td align="center">1.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00038.overview.png">coverage</a></td>
        <td><a href="calibration/37.unique_coverage.pdf">distribution</a></td>
        <td>contig00038</td>
        <td align="right">76,148</td>
        <td align="center">53.7</td>
        <td align="center">5.8</td>
        <td align="center">1.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00039.overview.png">coverage</a></td>
        <td><a href="calibration/38.unique_coverage.pdf">distribution</a></td>
        <td>contig00039</td>
        <td align="right">72,584</td>
        <td align="center">53.6</td>
        <td align="center">5.0</td>
        <td align="center">1.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00040.overview.png">coverage</a></td>
        <td><a href="calibration/39.unique_coverage.pdf">distribution</a></td>
        <td>contig00040</td>
        <td align="right">69,143</td>
        <td align="center">53.4</td>
        <td align="center">4.1</td>
        <td align="center">1.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00041.overview.png">coverage</a></td>
        <td><a href="calibration/40.unique_coverage.pdf">distribution</a></td>
        <td>contig00041</td>
        <td align="right">68,060</td>
        <td align="center">53.2</td>
        <td align="center">5.8</td>
        <td align="center">0.9%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00042.overview.png">coverage</a></td>
        <td><a href="calibration/41.unique_coverage.pdf">distribution</a></td>
        <td>contig00042</td>
        <td align="right">67,982</td>
        <td align="center">49.8</td>
        <td align="center">4.8</td>
        <td align="center">0.9%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00043.overview.png">coverage</a></td>
        <td><a href="calibration/42.unique_coverage.pdf">distribution</a></td>
        <td>contig00043</td>
        <td align="right">63,128</td>
        <td align="center">51.7</td>
        <td align="center">4.7</td>
        <td align="center">0.9%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00044.overview.png">coverage</a></td>
        <td><a href="calibration/43.unique_coverage.pdf">distribution</a></td>
        <td>contig00044</td>
        <td align="right">58,413</td>
        <td align="center">51.8</td>
        <td align="center">4.8</td>
        <td align="center">0.8%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00045.overview.png">coverage</a></td>
        <td><a href="calibration/44.unique_coverage.pdf">distribution</a></td>
        <td>contig00045</td>
        <td align="right">56,454</td>
        <td align="center">63.2</td>
        <td align="center">6.1</td>
        <td align="center">1.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00046.overview.png">coverage</a></td>
        <td><a href="calibration/45.unique_coverage.pdf">distribution</a></td>
        <td>contig00046</td>
        <td align="right">53,825</td>
        <td align="center">57.0</td>
        <td align="center">5.6</td>
        <td align="center">0.8%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00047.overview.png">coverage</a></td>
        <td><a href="calibration/46.unique_coverage.pdf">distribution</a></td>
        <td>contig00047</td>
        <td align="right">48,541</td>
        <td align="center">48.5</td>
        <td align="center">4.0</td>
        <td align="center">0.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00048.overview.png">coverage</a></td>
        <td><a href="calibration/47.unique_coverage.pdf">distribution</a></td>
        <td>contig00048</td>
        <td align="right">46,624</td>
        <td align="center">63.4</td>
        <td align="center">6.4</td>
        <td align="center">0.7%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00049.overview.png">coverage</a></td>
        <td><a href="calibration/48.unique_coverage.pdf">distribution</a></td>
        <td>contig00049</td>
        <td align="right">46,356</td>
        <td align="center">52.5</td>
        <td align="center">4.8</td>
        <td align="center">0.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00050.overview.png">coverage</a></td>
        <td><a href="calibration/49.unique_coverage.pdf">distribution</a></td>
        <td>contig00050</td>
        <td align="right">45,457</td>
        <td align="center">54.1</td>
        <td align="center">4.6</td>
        <td align="center">0.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00051.overview.png">coverage</a></td>
        <td><a href="calibration/50.unique_coverage.pdf">distribution</a></td>
        <td>contig00051</td>
        <td align="right">40,728</td>
        <td align="center">55.2</td>
        <td align="center">5.7</td>
        <td align="center">0.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00052.overview.png">coverage</a></td>
        <td><a href="calibration/51.unique_coverage.pdf">distribution</a></td>
        <td>contig00052</td>
        <td align="right">40,657</td>
        <td align="center">49.7</td>
        <td align="center">3.5</td>
        <td align="center">0.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00053.overview.png">coverage</a></td>
        <td><a href="calibration/52.unique_coverage.pdf">distribution</a></td>
        <td>contig00053</td>
        <td align="right">39,821</td>
        <td align="center">67.3</td>
        <td align="center">6.2</td>
        <td align="center">0.7%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00054.overview.png">coverage</a></td>
        <td><a href="calibration/53.unique_coverage.pdf">distribution</a></td>
        <td>contig00054</td>
        <td align="right">36,958</td>
        <td align="center">47.8</td>
        <td align="center">3.8</td>
        <td align="center">0.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00055.overview.png">coverage</a></td>
        <td><a href="calibration/54.unique_coverage.pdf">distribution</a></td>
        <td>contig00055</td>
        <td align="right">35,462</td>
        <td align="center">50.4</td>
        <td align="center">4.5</td>
        <td align="center">0.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00056.overview.png">coverage</a></td>
        <td><a href="calibration/55.unique_coverage.pdf">distribution</a></td>
        <td>contig00056</td>
        <td align="right">33,102</td>
        <td align="center">49.3</td>
        <td align="center">7.3</td>
        <td align="center">0.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00057.overview.png">coverage</a></td>
        <td><a href="calibration/56.unique_coverage.pdf">distribution</a></td>
        <td>contig00057</td>
        <td align="right">27,860</td>
        <td align="center">88.0</td>
        <td align="center">13.7</td>
        <td align="center">0.6%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00058.overview.png">coverage</a></td>
        <td><a href="calibration/57.unique_coverage.pdf">distribution</a></td>
        <td>contig00058</td>
        <td align="right">27,428</td>
        <td align="center">61.4</td>
        <td align="center">4.7</td>
        <td align="center">0.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00059.overview.png">coverage</a></td>
        <td><a href="calibration/58.unique_coverage.pdf">distribution</a></td>
        <td>contig00059</td>
        <td align="right">27,156</td>
        <td align="center">50.9</td>
        <td align="center">5.0</td>
        <td align="center">0.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00060.overview.png">coverage</a></td>
        <td><a href="calibration/59.unique_coverage.pdf">distribution</a></td>
        <td>contig00060</td>
        <td align="right">26,587</td>
        <td align="center">50.5</td>
        <td align="center">4.8</td>
        <td align="center">0.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00061.overview.png">coverage</a></td>
        <td><a href="calibration/60.unique_coverage.pdf">distribution</a></td>
        <td>contig00061</td>
        <td align="right">25,873</td>
        <td align="center">58.8</td>
        <td align="center">5.0</td>
        <td align="center">0.4%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00062.overview.png">coverage</a></td>
        <td><a href="calibration/61.unique_coverage.pdf">distribution</a></td>
        <td>contig00062</td>
        <td align="right">23,606</td>
        <td align="center">55.3</td>
        <td align="center">4.1</td>
        <td align="center">0.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00063.overview.png">coverage</a></td>
        <td><a href="calibration/62.unique_coverage.pdf">distribution</a></td>
        <td>contig00063</td>
        <td align="right">22,273</td>
        <td align="center">48.5</td>
        <td align="center">6.9</td>
        <td align="center">0.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00064.overview.png">coverage</a></td>
        <td><a href="calibration/63.unique_coverage.pdf">distribution</a></td>
        <td>contig00064</td>
        <td align="right">21,932</td>
        <td align="center">59.3</td>
        <td align="center">4.5</td>
        <td align="center">0.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00065.overview.png">coverage</a></td>
        <td><a href="calibration/64.unique_coverage.pdf">distribution</a></td>
        <td>contig00065</td>
        <td align="right">19,298</td>
        <td align="center">45.6</td>
        <td align="center">2.9</td>
        <td align="center">0.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00066.overview.png">coverage</a></td>
        <td><a href="calibration/65.unique_coverage.pdf">distribution</a></td>
        <td>contig00066</td>
        <td align="right">18,189</td>
        <td align="center">57.0</td>
        <td align="center">8.6</td>
        <td align="center">0.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00067.overview.png">coverage</a></td>
        <td><a href="calibration/66.unique_coverage.pdf">distribution</a></td>
        <td>contig00067</td>
        <td align="right">17,200</td>
        <td align="center">52.8</td>
        <td align="center">7.1</td>
        <td align="center">0.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00068.overview.png">coverage</a></td>
        <td><a href="calibration/67.unique_coverage.pdf">distribution</a></td>
        <td>contig00068</td>
        <td align="right">14,439</td>
        <td align="center">52.7</td>
        <td align="center">2.8</td>
        <td align="center">0.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00069.overview.png">coverage</a></td>
        <td><a href="calibration/68.unique_coverage.pdf">distribution</a></td>
        <td>contig00069</td>
        <td align="right">12,147</td>
        <td align="center">100.3</td>
        <td align="center">5.6</td>
        <td align="center">0.3%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00070.overview.png">coverage</a></td>
        <td><a href="calibration/69.unique_coverage.pdf">distribution</a></td>
        <td>contig00070</td>
        <td align="right">11,282</td>
        <td align="center">58.2</td>
        <td align="center">3.8</td>
        <td align="center">0.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00071.overview.png">coverage</a></td>
        <td><a href="calibration/70.unique_coverage.pdf">distribution</a></td>
        <td>contig00071</td>
        <td align="right">7,727</td>
        <td align="center">50.4</td>
        <td align="center">3.7</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00072.overview.png">coverage</a></td>
        <td><a href="calibration/71.unique_coverage.pdf">distribution</a></td>
        <td>contig00072</td>
        <td align="right">7,427</td>
        <td align="center">58.4</td>
        <td align="center">3.7</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00073.overview.png">coverage</a></td>
        <td><a href="calibration/72.unique_coverage.pdf">distribution</a></td>
        <td>contig00073</td>
        <td align="right">5,629</td>
        <td align="center">37.6</td>
        <td align="center">2.3</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00074.overview.png">coverage</a></td>
        <td><a href="calibration/73.unique_coverage.pdf">distribution</a></td>
        <td>contig00074</td>
        <td align="right">4,678</td>
        <td align="center">66.8</td>
        <td align="center">2.3</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00075.overview.png">coverage</a></td>
        <td><a href="calibration/74.unique_coverage.pdf">distribution</a></td>
        <td>contig00075</td>
        <td align="right">4,096</td>
        <td align="center">53.8</td>
        <td align="center">1.4</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00076.overview.png">coverage</a></td>
        <td><a href="calibration/75.unique_coverage.pdf">distribution</a></td>
        <td>contig00076</td>
        <td align="right">4,066</td>
        <td align="center">61.5</td>
        <td align="center">2.9</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00077.overview.png">coverage</a></td>
        <td><a href="calibration/76.unique_coverage.pdf">distribution</a></td>
        <td>contig00077</td>
        <td align="right">3,854</td>
        <td align="center">75.2</td>
        <td align="center">3.4</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00078.overview.png">coverage</a></td>
        <td><a href="calibration/77.unique_coverage.pdf">distribution</a></td>
        <td>contig00078</td>
        <td align="right">3,514</td>
        <td align="center">69.0</td>
        <td align="center">3.2</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00079.overview.png">coverage</a></td>
        <td><a href="calibration/78.unique_coverage.pdf">distribution</a></td>
        <td>contig00079</td>
        <td align="right">3,402</td>
        <td align="center">67.5</td>
        <td align="center">3.7</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00080.overview.png">coverage</a></td>
        <td><a href="calibration/79.unique_coverage.pdf">distribution</a></td>
        <td>contig00080</td>
        <td align="right">3,170</td>
        <td align="center">666.6</td>
        <td align="center">22.7</td>
        <td align="center">0.5%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00081.overview.png">coverage</a></td>
        <td><a href="calibration/80.unique_coverage.pdf">distribution</a></td>
        <td>contig00081</td>
        <td align="right">2,708</td>
        <td align="center">76.9</td>
        <td align="center">3.5</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00082.overview.png">coverage</a></td>
        <td><a href="calibration/81.unique_coverage.pdf">distribution</a></td>
        <td>contig00082</td>
        <td align="right">2,537</td>
        <td align="center">50.6</td>
        <td align="center">2.4</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00083.overview.png">coverage</a></td>
        <td><a href="calibration/82.unique_coverage.pdf">distribution</a></td>
        <td>contig00083</td>
        <td align="right">2,482</td>
        <td align="center">308.4</td>
        <td align="center">18.2</td>
        <td align="center">0.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00084.overview.png">coverage</a></td>
        <td><a href="calibration/83.unique_coverage.pdf">distribution</a></td>
        <td>contig00084</td>
        <td align="right">2,302</td>
        <td align="center">84.0</td>
        <td align="center">4.1</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00085.overview.png">coverage</a></td>
        <td><a href="calibration/84.unique_coverage.pdf">distribution</a></td>
        <td>contig00085</td>
        <td align="right">1,863</td>
        <td align="center">217.8</td>
        <td align="center">4.6</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00086.overview.png">coverage</a></td>
        <td><a href="calibration/85.unique_coverage.pdf">distribution</a></td>
        <td>contig00086</td>
        <td align="right">1,659</td>
        <td align="center">28.4</td>
        <td align="center">1.0</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00087.overview.png">coverage</a></td>
        <td><a href="calibration/86.unique_coverage.pdf">distribution</a></td>
        <td>contig00087</td>
        <td align="right">1,541</td>
        <td align="center">654.0</td>
        <td align="center">12.1</td>
        <td align="center">0.2%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00088.overview.png">coverage</a></td>
        <td><a href="calibration/87.unique_coverage.pdf">distribution</a></td>
        <td>contig00088</td>
        <td align="right">1,457</td>
        <td align="center">294.1</td>
        <td align="center">5.5</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00089.overview.png">coverage</a></td>
        <td><a href="calibration/88.unique_coverage.pdf">distribution</a></td>
        <td>contig00089</td>
        <td align="right">1,355</td>
        <td align="center">327.3</td>
        <td align="center">4.4</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00090.overview.png">coverage</a></td>
        <td><a href="calibration/89.unique_coverage.pdf">distribution</a></td>
        <td>contig00090</td>
        <td align="right">1,355</td>
        <td align="center">213.1</td>
        <td align="center">9.5</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00091.overview.png">coverage</a></td>
        <td><a href="calibration/90.unique_coverage.pdf">distribution</a></td>
        <td>contig00091</td>
        <td align="right">1,345</td>
        <td align="center">139.5</td>
        <td align="center">6.8</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00092.overview.png">coverage</a></td>
        <td><a href="calibration/91.unique_coverage.pdf">distribution</a></td>
        <td>contig00092</td>
        <td align="right">1,295</td>
        <td align="center">88.5</td>
        <td align="center">6.0</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00093.overview.png">coverage</a></td>
        <td><a href="calibration/92.unique_coverage.pdf">distribution</a></td>
        <td>contig00093</td>
        <td align="right">1,195</td>
        <td align="center">127.3</td>
        <td align="center">4.6</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00094.overview.png">coverage</a></td>
        <td><a href="calibration/93.unique_coverage.pdf">distribution</a></td>
        <td>contig00094</td>
        <td align="right">1,190</td>
        <td align="center">139.6</td>
        <td align="center">1.0</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00095.overview.png">coverage</a></td>
        <td><a href="calibration/94.unique_coverage.pdf">distribution</a></td>
        <td>contig00095</td>
        <td align="right">1,163</td>
        <td align="center">83.8</td>
        <td align="center">2.8</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00096.overview.png">coverage</a></td>
        <td><a href="calibration/95.unique_coverage.pdf">distribution</a></td>
        <td>contig00096</td>
        <td align="right">1,101</td>
        <td align="center">140.0</td>
        <td align="center">1.7</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00097.overview.png">coverage</a></td>
        <td><a href="calibration/96.unique_coverage.pdf">distribution</a></td>
        <td>contig00097</td>
        <td align="right">911</td>
        <td align="center">355.8</td>
        <td align="center">2.9</td>
        <td align="center">0.1%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00098.overview.png">coverage</a></td>
        <td><a href="calibration/97.unique_coverage.pdf">distribution</a></td>
        <td>contig00098</td>
        <td align="right">896</td>
        <td align="center">108.9</td>
        <td align="center">5.6</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00099.overview.png">coverage</a></td>
        <td><a href="calibration/98.unique_coverage.pdf">distribution</a></td>
        <td>contig00099</td>
        <td align="right">882</td>
        <td align="center">131.3</td>
        <td align="center">11.7</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00100.overview.png">coverage</a></td>
        <td><a href="calibration/99.unique_coverage.pdf">distribution</a></td>
        <td>contig00100</td>
        <td align="right">858</td>
        <td align="center">128.7</td>
        <td align="center">1.4</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00101.overview.png">coverage</a></td>
        <td><a href="calibration/100.unique_coverage.pdf">distribution</a></td>
        <td>contig00101</td>
        <td align="right">735</td>
        <td align="center">86.9</td>
        <td align="center">5.6</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00102.overview.png">coverage</a></td>
        <td><a href="calibration/101.unique_coverage.pdf">distribution</a></td>
        <td>contig00102</td>
        <td align="right">666</td>
        <td align="center">152.3</td>
        <td align="center">1.2</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00103.overview.png">coverage</a></td>
        <td><a href="calibration/102.unique_coverage.pdf">distribution</a></td>
        <td>contig00103</td>
        <td align="right">613</td>
        <td align="center">29.5</td>
        <td align="center">2.6</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00104.overview.png">coverage</a></td>
        <td><a href="calibration/103.unique_coverage.pdf">distribution</a></td>
        <td>contig00104</td>
        <td align="right">559</td>
        <td align="center">35.2</td>
        <td align="center">5.4</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00105.overview.png">coverage</a></td>
        <td><a href="calibration/104.unique_coverage.pdf">distribution</a></td>
        <td>contig00105</td>
        <td align="right">525</td>
        <td align="center">357.9</td>
        <td align="center">8.3</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00106.overview.png">coverage</a></td>
        <td><a href="calibration/105.unique_coverage.pdf">distribution</a></td>
        <td>contig00106</td>
        <td align="right">515</td>
        <td align="center">37.6</td>
        <td align="center">2.0</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr>
        <td><a href="evidence/contig00107.overview.png">coverage</a></td>
        <td><a href="calibration/106.unique_coverage.pdf">distribution</a></td>
        <td>contig00107</td>
        <td align="right">503</td>
        <td align="center">97.7</td>
        <td align="center">1.3</td>
        <td align="center">0.0%</td>
        <td>na</td>
    </tr>
    <tr class="highlight_table_row">
        <td></td>
        <td></td>
        <td><b>total</b></td>
        <td align="right"><b>6,994,048</b></td>
        <td></td>
        <td></td>
        <td align="center">100.0%</td>
        <td></td>
    </tr>
</table>
<p><b>fit dispersion</b> is the ratio of the variance to the mean for the negative binomial fit. It is =1 for Poisson and >1 for over-dispersed data.
<h2>New Junction Evidence</h2>
<p>
<h3>Junction Candidates Tested</h3>
<p>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th align="center">option</th>
        <th align="center">limit</th>
        <th align="center">actual</th>
    </tr>
    <tr>
        <td>Number of alignment pairs examined for constructing junction candidates</td>
        <td align="center">&le;&nbsp;100000</td>
        <td align="center">3005</td>
    </tr>
    <tr>
        <td>Coverage evenness (position-hash) score of junction candidates</td>
        <td align="center">&ge;&nbsp;2</td>
        <td align="center">&ge;&nbsp;2</td>
    </tr>
    <tr>
        <td>Test this many junction candidates (n). May be smaller if not enough passed the coverage evenness threshold</td>
        <td align="center">100&nbsp;&le;&nbsp;n&nbsp;&le;&nbsp;5000</td>
        <td align="center">492</td>
    </tr>
    <tr>
        <td>Total length of all junction candidates (factor times the reference genome length)</td>
        <td align="center">&le;&nbsp;0.1</td>
        <td align="center">0.021</td>
    </tr>
</table>
<h3>Junction Skew Score Calculation</h3>
<p>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th>reference sequence</th>
        <th>pr(no read start)</th>
    </tr>
    <tr>
        <td>contig00001</td>
        <td align="center">0.85829</td>
    </tr>
    <tr>
        <td>contig00002</td>
        <td align="center">0.86661</td>
    </tr>
    <tr>
        <td>contig00003</td>
        <td align="center">0.86320</td>
    </tr>
    <tr>
        <td>contig00004</td>
        <td align="center">0.85660</td>
    </tr>
    <tr>
        <td>contig00005</td>
        <td align="center">0.85887</td>
    </tr>
    <tr>
        <td>contig00006</td>
        <td align="center">0.86470</td>
    </tr>
    <tr>
        <td>contig00007</td>
        <td align="center">0.85690</td>
    </tr>
    <tr>
        <td>contig00008</td>
        <td align="center">0.85232</td>
    </tr>
    <tr>
        <td>contig00009</td>
        <td align="center">0.85514</td>
    </tr>
    <tr>
        <td>contig00010</td>
        <td align="center">0.85548</td>
    </tr>
    <tr>
        <td>contig00011</td>
        <td align="center">0.84887</td>
    </tr>
    <tr>
        <td>contig00012</td>
        <td align="center">0.83221</td>
    </tr>
    <tr>
        <td>contig00013</td>
        <td align="center">0.86653</td>
    </tr>
    <tr>
        <td>contig00014</td>
        <td align="center">0.83920</td>
    </tr>
    <tr>
        <td>contig00015</td>
        <td align="center">0.84312</td>
    </tr>
    <tr>
        <td>contig00016</td>
        <td align="center">0.86490</td>
    </tr>
    <tr>
        <td>contig00017</td>
        <td align="center">0.85631</td>
    </tr>
    <tr>
        <td>contig00018</td>
        <td align="center">0.86264</td>
    </tr>
    <tr>
        <td>contig00019</td>
        <td align="center">0.84045</td>
    </tr>
    <tr>
        <td>contig00020</td>
        <td align="center">0.86209</td>
    </tr>
    <tr>
        <td>contig00021</td>
        <td align="center">0.85413</td>
    </tr>
    <tr>
        <td>contig00022</td>
        <td align="center">0.86025</td>
    </tr>
    <tr>
        <td>contig00023</td>
        <td align="center">0.85873</td>
    </tr>
    <tr>
        <td>contig00024</td>
        <td align="center">0.85982</td>
    </tr>
    <tr>
        <td>contig00025</td>
        <td align="center">0.85759</td>
    </tr>
    <tr>
        <td>contig00026</td>
        <td align="center">0.84626</td>
    </tr>
    <tr>
        <td>contig00027</td>
        <td align="center">0.85473</td>
    </tr>
    <tr>
        <td>contig00028</td>
        <td align="center">0.82293</td>
    </tr>
    <tr>
        <td>contig00029</td>
        <td align="center">0.86858</td>
    </tr>
    <tr>
        <td>contig00030</td>
        <td align="center">0.84265</td>
    </tr>
    <tr>
        <td>contig00031</td>
        <td align="center">0.85844</td>
    </tr>
    <tr>
        <td>contig00032</td>
        <td align="center">0.86300</td>
    </tr>
    <tr>
        <td>contig00033</td>
        <td align="center">0.85973</td>
    </tr>
    <tr>
        <td>contig00034</td>
        <td align="center">0.82354</td>
    </tr>
    <tr>
        <td>contig00035</td>
        <td align="center">0.83706</td>
    </tr>
    <tr>
        <td>contig00036</td>
        <td align="center">0.85983</td>
    </tr>
    <tr>
        <td>contig00037</td>
        <td align="center">0.85301</td>
    </tr>
    <tr>
        <td>contig00038</td>
        <td align="center">0.86059</td>
    </tr>
    <tr>
        <td>contig00039</td>
        <td align="center">0.85944</td>
    </tr>
    <tr>
        <td>contig00040</td>
        <td align="center">0.85666</td>
    </tr>
    <tr>
        <td>contig00041</td>
        <td align="center">0.85882</td>
    </tr>
    <tr>
        <td>contig00042</td>
        <td align="center">0.86274</td>
    </tr>
    <tr>
        <td>contig00043</td>
        <td align="center">0.85742</td>
    </tr>
    <tr>
        <td>contig00044</td>
        <td align="center">0.85872</td>
    </tr>
    <tr>
        <td>contig00045</td>
        <td align="center">0.83019</td>
    </tr>
    <tr>
        <td>contig00046</td>
        <td align="center">0.84546</td>
    </tr>
    <tr>
        <td>contig00047</td>
        <td align="center">0.86546</td>
    </tr>
    <tr>
        <td>contig00048</td>
        <td align="center">0.83940</td>
    </tr>
    <tr>
        <td>contig00049</td>
        <td align="center">0.86164</td>
    </tr>
    <tr>
        <td>contig00050</td>
        <td align="center">0.85319</td>
    </tr>
    <tr>
        <td>contig00051</td>
        <td align="center">0.85192</td>
    </tr>
    <tr>
        <td>contig00052</td>
        <td align="center">0.85986</td>
    </tr>
    <tr>
        <td>contig00053</td>
        <td align="center">0.81889</td>
    </tr>
    <tr>
        <td>contig00054</td>
        <td align="center">0.86135</td>
    </tr>
    <tr>
        <td>contig00055</td>
        <td align="center">0.86199</td>
    </tr>
    <tr>
        <td>contig00056</td>
        <td align="center">0.86768</td>
    </tr>
    <tr>
        <td>contig00057</td>
        <td align="center">0.80505</td>
    </tr>
    <tr>
        <td>contig00058</td>
        <td align="center">0.83983</td>
    </tr>
    <tr>
        <td>contig00059</td>
        <td align="center">0.86379</td>
    </tr>
    <tr>
        <td>contig00060</td>
        <td align="center">0.86334</td>
    </tr>
    <tr>
        <td>contig00061</td>
        <td align="center">0.84068</td>
    </tr>
    <tr>
        <td>contig00062</td>
        <td align="center">0.85421</td>
    </tr>
    <tr>
        <td>contig00063</td>
        <td align="center">0.86945</td>
    </tr>
    <tr>
        <td>contig00064</td>
        <td align="center">0.84177</td>
    </tr>
    <tr>
        <td>contig00065</td>
        <td align="center">0.85879</td>
    </tr>
    <tr>
        <td>contig00066</td>
        <td align="center">0.86401</td>
    </tr>
    <tr>
        <td>contig00067</td>
        <td align="center">0.85841</td>
    </tr>
    <tr>
        <td>contig00068</td>
        <td align="center">0.85650</td>
    </tr>
    <tr>
        <td>contig00069</td>
        <td align="center">0.77772</td>
    </tr>
    <tr>
        <td>contig00070</td>
        <td align="center">0.84864</td>
    </tr>
    <tr>
        <td>contig00071</td>
        <td align="center">0.85454</td>
    </tr>
    <tr>
        <td>contig00072</td>
        <td align="center">0.84230</td>
    </tr>
    <tr>
        <td>contig00073</td>
        <td align="center">0.86436</td>
    </tr>
    <tr>
        <td>contig00074</td>
        <td align="center">0.82818</td>
    </tr>
    <tr>
        <td>contig00075</td>
        <td align="center">0.85093</td>
    </tr>
    <tr>
        <td>contig00076</td>
        <td align="center">0.84380</td>
    </tr>
    <tr>
        <td>contig00077</td>
        <td align="center">0.82106</td>
    </tr>
    <tr>
        <td>contig00078</td>
        <td align="center">0.82462</td>
    </tr>
    <tr>
        <td>contig00079</td>
        <td align="center">0.82861</td>
    </tr>
    <tr>
        <td>contig00080</td>
        <td align="center">0.35180</td>
    </tr>
    <tr>
        <td>contig00081</td>
        <td align="center">0.82035</td>
    </tr>
    <tr>
        <td>contig00082</td>
        <td align="center">0.87091</td>
    </tr>
    <tr>
        <td>contig00083</td>
        <td align="center">0.54856</td>
    </tr>
    <tr>
        <td>contig00084</td>
        <td align="center">0.80864</td>
    </tr>
    <tr>
        <td>contig00085</td>
        <td align="center">0.62006</td>
    </tr>
    <tr>
        <td>contig00086</td>
        <td align="center">0.88556</td>
    </tr>
    <tr>
        <td>contig00087</td>
        <td align="center">0.39754</td>
    </tr>
    <tr>
        <td>contig00088</td>
        <td align="center">0.56687</td>
    </tr>
    <tr>
        <td>contig00089</td>
        <td align="center">0.46878</td>
    </tr>
    <tr>
        <td>contig00090</td>
        <td align="center">0.62645</td>
    </tr>
    <tr>
        <td>contig00091</td>
        <td align="center">0.71725</td>
    </tr>
    <tr>
        <td>contig00092</td>
        <td align="center">0.78177</td>
    </tr>
    <tr>
        <td>contig00093</td>
        <td align="center">0.73347</td>
    </tr>
    <tr>
        <td>contig00094</td>
        <td align="center">0.72143</td>
    </tr>
    <tr>
        <td>contig00095</td>
        <td align="center">0.79708</td>
    </tr>
    <tr>
        <td>contig00096</td>
        <td align="center">0.71220</td>
    </tr>
    <tr>
        <td>contig00097</td>
        <td align="center">0.49759</td>
    </tr>
    <tr>
        <td>contig00098</td>
        <td align="center">0.77958</td>
    </tr>
    <tr>
        <td>contig00099</td>
        <td align="center">0.78005</td>
    </tr>
    <tr>
        <td>contig00100</td>
        <td align="center">0.75350</td>
    </tr>
    <tr>
        <td>contig00101</td>
        <td align="center">0.82381</td>
    </tr>
    <tr>
        <td>contig00102</td>
        <td align="center">0.71354</td>
    </tr>
    <tr>
        <td>contig00103</td>
        <td align="center">0.91429</td>
    </tr>
    <tr>
        <td>contig00104</td>
        <td align="center">0.91950</td>
    </tr>
    <tr>
        <td>contig00105</td>
        <td align="center">0.57412</td>
    </tr>
    <tr>
        <td>contig00106</td>
        <td align="center">0.89877</td>
    </tr>
    <tr>
        <td>contig00107</td>
        <td align="center">0.80671</td>
    </tr>
</table>
<p><b>pr(no read start)</b> is the probability that there will not be an aligned read whose first base matches a given position on a given strand.
<h3>Final Junction Predictions</h3>
<p>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th align="center">option</th>
        <th align="center">value</th>
    </tr>
    <tr>
        <td>Coverage evenness (position-hash) score of predicted junctions must be</td>
        <td align="center">&ge;&nbsp;3</td>
    </tr>
    <tr>
        <td>Skew score of predicted junction (&minus;log10 probability of unusual coverage evenness) must be</td>
        <td align="center">&le;&nbsp;3</td>
    </tr>
    <tr>
        <td>Number of bases that at least one read must overlap each uniquely aligned side of a predicted junction</td>
        <td align="center">&ge;&nbsp;1</td>
    </tr>
</table>
<h2>Read Alignment Evidence</h2>
<p>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th>option</th>
        <th>value</th>
    </tr>
    <tr>
        <td>Mode</td>
        <td>Consensus/Mixed Base</td>
    </tr>
    <tr>
        <td>Ploidy</td>
        <td>1 (haploid)</td>
    </tr>
    <tr>
        <td>Consensus mutation E-value cutoff</td>
        <td>10</td>
    </tr>
    <tr>
        <td>Consensus frequency cutoff</td>
        <td>0.8</td>
    </tr>
    <tr>
        <td>Consensus minimum variant coverage each strand</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Consensus minimum total coverage each strand</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Consensus minimum variant coverage</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Consensus minimum total coverage</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Polymorphism E-value cutoff</td>
        <td>10</td>
    </tr>
    <tr>
        <td>Polymorphism frequency cutoff</td>
        <td>0.2</td>
    </tr>
    <tr>
        <td>Polymorphism minimum variant coverage each strand</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Polymorphism minimum total coverage each strand</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Polymorphism minimum variant coverage</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Polymorphism minimum total coverage</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Polymorphism bias cutoff</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Predict indel polymorphisms</td>
        <td>YES</td>
    </tr>
    <tr>
        <td>Skip indel polymorphisms in homopolymers runs of</td>
        <td>OFF</td>
    </tr>
    <tr>
        <td>Skip base substitutions when they create a homopolymer flanked on each side by</td>
        <td>OFF</td>
    </tr>
</table>
<p>
<h2>Software Versions</h2>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th>program</th>
        <th>version</th>
    </tr>
    <tr>
        <td>bowtie2</td>
        <td>2.3.2</td>
    </tr>
    <tr>
        <td>R</td>
        <td>3.2.1</td>
    </tr>
</table>
<p>
<h2>Execution Times</h2>
<table border="0" cellspacing="1" cellpadding="5">
    <tr>
        <th>step</th>
        <th>start</th>
        <th>end</th>
        <th>elapsed</th>
    </tr>
    <tr>
        <td>Read and reference sequence file input</td>
        <td>00:52:46&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>00:53:17&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>31&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Read alignment to reference genome</td>
        <td>00:53:17&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:01:03&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>7&nbsp;minutes&nbsp;46&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Preprocessing alignments for candidate junction identification</td>
        <td>01:01:03&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:01:43&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>40&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Preliminary analysis of coverage distribution</td>
        <td>01:01:43&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:04:52&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>3&nbsp;minutes&nbsp;9&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Identifying junction candidates</td>
        <td>01:04:52&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:05:00&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>8&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Re-alignment to junction candidates</td>
        <td>01:05:00&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:07:02&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>2&nbsp;minutes&nbsp;2&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Resolving best read alignments</td>
        <td>01:07:02&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:08:14&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>1&nbsp;minute&nbsp;12&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Creating BAM files</td>
        <td>01:08:14&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:10:26&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>2&nbsp;minutes&nbsp;12&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Tabulating error counts</td>
        <td>01:10:26&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:10:59&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>33&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Re-calibrating base error rates</td>
        <td>01:10:59&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:11:36&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>37&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Examining read alignment evidence</td>
        <td>01:11:36&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:18:58&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>7&nbsp;minutes&nbsp;22&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Polymorphism statistics</td>
        <td>01:18:58&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:18:58&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>0&nbsp;seconds</td>
    </tr>
    <tr>
        <td>Output</td>
        <td>01:18:58&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>01:19:46&nbsp;28&nbsp;Jul&nbsp;2019</td>
        <td>48&nbsp;seconds</td>
    </tr>
    <tr class="highlight_table_row">
        <td colspan="3"><b>Total</b></td>
        <td><b>27&nbsp;minutes&nbsp;0&nbsp;seconds</b></td>
    </tr>
</table>
</body>
</html>